CMakeLists.txt

```diff
@@ -28,6 +28,13 @@ SET(CMAKE_COLOR_MAKEFILE ON)
 # set path to additional CMake modules
 SET(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH})
 
+# Configure CCache if available
+find_program(CCACHE_FOUND ccache)
+if(CCACHE_FOUND)
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
+endif(CCACHE_FOUND)
```

Comment:
@lbartoletti we have very long build time on Travis, I wonder if this is the origin. Do you think it might conflict with the fact that `ccache` was already set [here](https://github.com/qgis/QGIS/blob/master/.docker/qgis.dockerfile#L11-L12) in the Dockerfile: 
```
ENV CC=/usr/lib/ccache/clang
ENV CXX=/usr/lib/ccache/clang++
```
ping @m-kuhn

Reply:
not sure what's going on. maybe we should clean the cache on travis and start over with a clean one? and if that doesn't help add an option to disable this logic?

Reply:
opened a PR to revert and check if this really is the responsible commit https://github.com/qgis/QGIS/pull/6741

Reply:
Not sure about it. I haven't errors at home ( c ).
But if it conflicts, you can use the second option as mentioned in my PR:

```
option(USE_CCACHE "Use ccache" OFF)
if (USE_CCACHE)
    find_program(CCACHE_FOUND ccache)
    if(CCACHE_FOUND)
        set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
        set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
    endif(CCACHE_FOUND)
endif()
```